documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -310,106 +103,134 @@ Controller services can be configured in three different ways:
+use the built-in Password Auth Handler that supports "Basic" authentication, you may supply the credentials as shown below.
 
-  ```
+  ```java
   ClientConfig clientConfig = ClientConfig.builder()
-                .controllerURI("tls://<DNS-NAME-OR-IP>:9090")
-                .trustStore("/etc/secrets/ca-cert")
+                .controllerURI("tls://<dns-name-or-ip>:9090")
+                .trustStore("/etc/secrets/ca-cert.crt")
                 .credentials(new DefaultCredentials("changeit", "marketinganaylticsapp"))
                 .build();
   ```
 
 #### Server Hostname Verification
 
-Hostname verification during TLS communications verifies that the DNS name to which the client connects matches the hostname specified in either of the following fields in the server's certificate:
+For client's server hostname verification to succeed during TLS handshake, the hostname/IP address it uses for accessing the
+server must match one of the following in the server's certificate:
 
 * Common Name (`CN`) in the certificate's `Subject` field
-* One of the `Subject Alternative Names` field entries
+* One of the `Subject Alternative Names` (SAN) field entries
 
-If the server certificates have a hostname assigned, you have used IP addresses as endpoints for the services, and those hostnames are not accessible from
-the client nodes, you might need to add mappings of
-IP addresses and DNS/Host names in the client-side operating system hosts file.
+Even if the server listens on the loopback address `127.0.0.1` and its certificate is assigned to `localhost`, hostname 
+verification will fail if the client attempts to access the server via `127.0.0.1`. For the verification to pass, 
+the client must access the server using the hostname assigned on the certificate `localhost` and have a hosts file entry 
+that maps `localhost` to `127.0.0.1` (which is usually already there). 
 
-Alternatively, you may disable hostname verification by invoking `validateHostName(false)` of the ClientConfig builder. It is strongly recommended to avoid disabling hostname verification for production clusters.
+Similarly, if the server certificate is assigned to a non-routable hostname on the network (say, `controller01.pravega.io`),
+you might need to add an IP address and DNS/host name mapping in the client's operating system hosts file. 
 
-### Having TLS and Auth Take Effect
+```
+10.243.221.34 controller01.pravega.io
+```
+
+If you are reusing preexisting certificate for development/testing on new hosts, you might need to disable hostname 
+verification. To do so, call `validateHostName(false)` of the ClientConfig builder, as shown below. Never disable 
+hostname verification in production.
 
-To ensure TLS and auth parameters take effect, all the services on the server-side need to be restarted.
-Existing client applications will need to be restarted as well, after they are reconfigured for TLS and auth.
+  ```java
+  ClientConfig clientConfig = ClientConfig.builder()
```

Comment:
In which file is the security configuration of clients being discussed in detail?

Reply:
We don't have a document for it yet, but it is shown here:  https://github.com/pravega/pravega/blob/master/documentation/src/docs/security/securing-standalone-mode-cluster.md.

Reply:
I've created a new issue for documenting client-side configurations.